library/Fission/Web/Heroku/Provision.hs

```diff
@@ -42,27 +43,31 @@ create Request {uuid, region} = do
   let username = Text.pack (UUID.toString uuid)
   secret <- Random.alphaNum 50
 
-  User.createWithHeroku uuid region username secret >>= \case
-    Left err ->
-      Web.Err.throw err
+  secret
+    |> User.createWithHeroku uuid region username
+    |> runDBNow
+    |> bind Web.Err.ensure
+    |> bind \userID -> do
+        Web.Host url' <- getHost
+        ipfsPeers     <- getIPFSPeers
 
-    Right userID -> do
-      Web.Host url' <- getHost
-      ipfsPeers     <- getExternalAddress >>= \case
-        Right peers' ->
-          pure peers'
+        return Provision
+          { id      = userID
+          , peers   = ipfsPeers
+          , message = "Successfully provisioned Interplanetary Fission!"
+          , config  = User.Provision
+            { username = username
+            , password = Secret secret
+            , url      = url'
+            }
+          }
```

Comment:
I don't love seeing...

```haskell
|> bind foo
|> bind bar
```

...but the good news about being in a pure lazy language is that the compiler _will_ rewrite this as:

```haskell
|> bind \x -> 
  foo (bar x)
```

Having a huge string of `bind...bind...bind...bind` would be annoying, so maybe let's limit it to like... 2 or 3 max?

Reply:
makes sense to me, & I think it reads fine here